### DISCUSSION TITLE
I don't know why this procsee wrong

### DISCUSSION THREAD

**Discussion by sinyoung-park** [open-mmlab/mmclassification#1696] [Q&A]
Hi, I using MMpretrain. and i use Resnet34 and i modified it. anyway, I was using it well until 3 days ago, when suddenly I ran train.py and got an error. TypeError: class `CustomDataset` in mmpretrain/datasets/custom.py: BaseDataset.__init__() got an unexpected keyword argument 'split'. i don't torch any other code. and i trying to find error. but the base_dataset.py and custom.py code not have problem. how can i fix? Did something happen to your split in the meantime?

**Comment by Ezra-Yu** [comment]
class `CustomDataset` don't have a parameter of `split`. please refer to this [doc](https://mmpretrain.readthedocs.io/en/latest/user_guides/dataset_prepare.html).

**Reply by sinyoung-park** [reply]
i  read refer, but i don't think fix my problem.  please help me. 
the custom.py is
 def__int__( is not have split.  @DATASETS.register_module()
class CustomDataset(BaseDataset):
    def __init__(
        self,
        data_root: str = '',
        data_prefix: Union[str, dict] = '',
        with_label: bool = True,
        extensions: Sequence[str] = ('.jpg', '.jpeg', '.png', '.ppm', '.bmp', '.pgm', '.tif'),
        metainfo: Optional[dict] = None,
        lazy_init: bool = False,
        **kwargs
    ):
        assert (data_prefix or data_root), 'One of `data_root` and `data_prefix` must be specified.'

        self.extensions = tuple(set([i.lower() for i in extensions]))
        self.with_label = with_label

        super().__init__(
            data_root=data_root,
            data_prefix=data_prefix,
            metainfo=metainfo,
            lazy_init=True,
            **kwargs
        )

        # Full initialize the dataset.
        if not lazy_init:
            self.full_init()
and my config file is
%%writefile configs/resnet/resnet34_8xb32_in1k_custom.py
_base_ = [
    '../_base_/models/resnet34.py', '../_base_/datasets/imagenet_bs32.py',
    '../_base_/schedules/imagenet_bs256.py', '../_base_/default_runtime.py'
]



# model settings
model = dict(
    type='ImageClassifier',
    backbone=dict(
        type='ResNet',
        depth=34,
        num_stages=4,
        out_indices=(3, ),
        style='pytorch'),
    neck=dict(type='GlobalAveragePooling'),
    head=dict(
        type='LinearClsHead',
        num_classes=5,
        in_channels=512,
        loss=dict(type='CrossEntropyLoss', loss_weight=0.8),
        topk=(1, 5),
    ))

# dataset settings
dataset_type = 'CustomDataset',
data_preprocessor = dict(
    num_classes=5,
    # RGB format normalization parameters
    mean=[123.675, 116.28, 103.53],
    std=[58.395, 57.12, 57.375],
    # convert image from BGR to RGB
    to_rgb=True,
)

train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='RandomResizedCrop', scale=224),
    dict(type='RandomFlip', prob=0.5, direction='horizontal'),
    dict(type='PackInputs'),
]

test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='ResizeEdge', scale=256, edge='short'),
    dict(type='CenterCrop', crop_size=224),
    dict(type='PackInputs'),
]

train_dataloader = dict(
    batch_size=16,
    num_workers=5,
    dataset=dict(
        type='CustomDataset',
        data_root='/content/mmpretrain/pcldata/potentialcore230522',
        ann_file='/content/mmpretrain/pcldata/potentialcore230522/train.txt',
        data_prefix='train_set',
        with_label=True,
        classes=['8slm','12slm','16slm','20slm','24slm'],
        pipeline=train_pipeline),
    sampler=dict(type='DefaultSampler', shuffle=True),
)

val_dataloader = dict(
    batch_size=16,
    num_workers=5,
    dataset=dict(
        type=dataset_type,
        data_root='/content/mmpretrain/pcldata/potentialcore230522',
        ann_file='/content/mmpretrain/pcldata/potentialcore230522/val.txt',
        data_prefix='val_set',
        with_label=True,
        classes=['8slm','12slm','16slm','20slm','24slm'],
        pipeline=test_pipeline),
    sampler=dict(type='DefaultSampler', shuffle=True),
)


test_dataloader = dict(
    batch_size=16,
    num_workers=5,
    dataset=dict(
        type=dataset_type,
        data_root='/content/mmpretrain/pcldata/potentialcore230522',
        ann_file='/content/mmpretrain/pcldata/potentialcore230522/test.txt',
        data_prefix='test_set',
        with_label=True,
        classes=['8slm','12slm','16slm','20slm','24slm'],
        pipeline=test_pipeline),
    sampler=dict(type='DefaultSampler', shuffle=False),
)
val_evaluator = dict(type='Accuracy', topk=(1, 5))


# optimizer
optim_wrapper = dict(
    optimizer=dict(type='SGD', lr=0.025, momentum=0.9, weight_decay=0.0001,dampening=0.001))

# learning policy
param_scheduler = dict(
    type='MultiStepLR', by_epoch=True, milestones=[10, 20, 30], gamma=0.1)

# train, val, test setting
train_cfg = dict(by_epoch=True, max_epochs=25, val_interval=1)
val_cfg = dict()
test_cfg = dict()

# NOTE: `auto_scale_lr` is for automatically scaling LR,
# based on the actual training batch size.
auto_scale_lr = dict(base_batch_size=256)

# defaults to use registries in mmpretrain
default_scope = 'mmpretrain'

# configure default hooks
default_hooks = dict(
    # record the time of every iteration.
    timer=dict(type='IterTimerHook'),

    # print log every 100 iterations.
    logger=dict(type='LoggerHook', interval=100),

    # enable the parameter scheduler.
    param_scheduler=dict(type='ParamSchedulerHook'),

    # save checkpoint per epoch.
    checkpoint=dict(type='CheckpointHook', interval=1),

    # set sampler seed in distributed evrionment.
    sampler_seed=dict(type='DistSamplerSeedHook'),

    # validation results visualization, set True to enable it.
    visualization=dict(type='VisualizationHook', enable=False),
)

# configure environment
env_cfg = dict(
    # whether to enable cudnn benchmark
    cudnn_benchmark=False,

    # set multi process parameters
    mp_cfg=dict(mp_start_method='fork', opencv_num_threads=0),

    # set distributed parameters
    dist_cfg=dict(backend='nccl'),
)

# set visualizer
vis_backends = [dict(type='LocalVisBackend')]
visualizer = dict(type='UniversalVisualizer', vis_backends=vis_backends)

# set log level
log_level = 'INFO'

# load from which checkpoint
load_from = None

# whether to resume training from the loaded checkpoint
resume = False

# Defaults to use random seed and disable `deterministic`
randomness = dict(seed=1, deterministic=True)
terministic=True)

**Reply by sinyoung-park** [reply]
i solve this problem. the error code is cause the split=' '  is not exist the custom.py class CustomDataset. __init__  Why is this happening? i don't think so. Anyway, thanks for the answer

**Reply by Leezp99** [reply]
how can you solve this problem? I meet this question, too.

**Reply by sinyoung-park** [reply]
insert split=' ' in to dataset/custom.py like this 

  def __init__(self,
                 data_root: str = '',
                 data_prefix: Union[str, dict] = '',
                 ann_file: str = '',
                 split=' ',
                 with_label=True,
                 extensions: Sequence[str] = ('.jpg', '.jpeg', '.png', '.ppm',
                                              '.bmp', '.pgm', '.tif'),
                 metainfo: Optional[dict] = None,
                 lazy_init: bool = False,
                 **kwargs):
        assert (ann_file or data_prefix or data_root), \
            'One of `ann_file`, `data_root` and `data_prefix` must '\
            'be specified.'

        self.extensions = tuple(set([i.lower() for i in extensions]))
        self.with_label = with_label
and you can solve this problem

**Reply by Julan-zy** [reply]
> insert split=' ' in to dataset/custom.py like this
> 
> def **init**(self, data_root: str = '', data_prefix: Union[str, dict] = '', ann_file: str = '', split=' ', with_label=True, extensions: Sequence[str] = ('.jpg', '.jpeg', '.png', '.ppm', '.bmp', '.pgm', '.tif'), metainfo: Optional[dict] = None, lazy_init: bool = False, **kwargs): assert (ann_file or data_prefix or data_root), 'One of `ann_file`, `data_root` and `data_prefix` must ' 'be specified.'
> 
> ```
>     self.extensions = tuple(set([i.lower() for i in extensions]))
>     self.with_label = with_label
> ```
> 
> and you can solve this problem

Hello! I meet this question, too. I added the line "split=' '," as you did, but a new error message appears:
Traceback (most recent call last):
  File "tools/train.py", line 162, in <module>
    main()
  File "tools/train.py", line 158, in main
    runner.train()
  File "C:\Users\Julan\anaconda3\envs\openmmlab\lib\site-packages\mmengine\runner\runner.py", line 1733, in train
    self.optim_wrapper = self.build_optim_wrapper(self.optim_wrapper)
  File "C:\Users\Julan\anaconda3\envs\openmmlab\lib\site-packages\mmengine\runner\runner.py", line 1114, in build_optim_wrapper
    return build_optim_wrapper(self.model, optim_wrapper)
  File "C:\Users\Julan\anaconda3\envs\openmmlab\lib\site-packages\mmengine\optim\optimizer\builder.py", line 210, in build_optim_wrapper
    optim_wrapper = optim_wrapper_constructor(model)
  File "C:\Users\Julan\anaconda3\envs\openmmlab\lib\site-packages\mmengine\optim\optimizer\default_constructor.py", line 318, in __call__
    optimizer = OPTIMIZERS.build(optimizer_cfg)
  File "C:\Users\Julan\anaconda3\envs\openmmlab\lib\site-packages\mmengine\registry\registry.py", line 570, in build
    return self.build_func(cfg, *args, **kwargs, registry=self)
  File "C:\Users\Julan\anaconda3\envs\openmmlab\lib\site-packages\mmengine\registry\build_functions.py", line 121, in build_from_cfg
    obj = obj_cls(**args)  # type: ignore
TypeError: __init__() got an unexpected keyword argument 'momentum'
